lib/xapi/implementation.rb

```diff
@@ -15,29 +15,32 @@ def initialize(track_id, repo, problem, root)
       @track_id = track_id
       @repo = repo
       @problem = problem
-      @root = root
+      @root = Pathname.new(root)
     end
 
     def exists?
       File.exist?(dir)
     end
 
     def files
-      @files ||= Hash[paths.map {|path|
-        [filename(path), File.read(path)]
+      @files ||= Hash[FileBundle.paths(dir, IGNORE).map {|path|
+        [path.relative_path_from(dir).to_s, File.read(path)]
       }].merge("README.md" => readme)
     end
 
     def zip
-      @zip ||= create_zip
+      @zip ||= FileBundle.create_zip(dir, FileBundle.paths(dir, IGNORE)) do |io|
+        io.put_next_entry('README.md')
+        io.print readme
```

Comment:
Since we now have a `FileBundle` class, is there a way we could move this knowledge about how to add a new file to the bundle into it?
It could be nice to be able to do something like:
`zip.add_file( name: 'README.md', content: readme )`
And not have the `Implementation` class knowing anything about the internals of a FileBundle